mlflow/models/utils.py

```diff
@@ -858,6 +871,78 @@ def _is_scalar(x):
         return _enforce_unnamed_col_schema(pf_input, input_schema)
 
 
+def _enforce_datatype(data: Any, dtype: DataType):
+    if not isinstance(dtype, DataType):
+        raise MlflowException(f"Expected dtype to be DataType, got {type(dtype).__name__}")
+    # we do not convert data type for users
+    if (
+        dtype == DataType.boolean
+        and DataType.is_boolean(data)
+        or dtype == DataType.integer
+        and DataType.is_integer(data)
```

Comment:
```suggestion
        (dtype == DataType.boolean and DataType.is_boolean(data))
        or (dtype == DataType.integer and DataType.is_integer(data))
```

Is it possible to add `()` for better readability?

Reply:
Reused `_enforce_mlflow_datatype` .